src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
```

Comment:
[FUNCTION NAMING] - The `hide` method should only set the `hidden` property to `true`, but currently, it toggles the value. This could lead to unexpected behavior where calling `hide` on an already hidden herbivore would unhide it, which is not the desired functionality.